packages/loader/driver-utils/src/treeConversions.ts

```diff
@@ -0,0 +1,62 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Uint8ArrayToString,unreachableCase} from "@fluidframework/common-utils";
+import { AttachmentTreeEntry, BlobTreeEntry, TreeTreeEntry } from "@fluidframework/protocol-base";
+import * as api from "@fluidframework/protocol-definitions";
+
+/**
+ * Converts ISummaryTree to ITree format.
+ * @param summaryTree - summary tree in ISummaryTree format
+ */
+ export function convertSummaryTreeToSnapshotITree( summaryTree: api.ISummaryTree ): api.ITree {
```

Comment:
We already have a `convertSummaryTreeToITree` function in `summaryUtils.ts`. Does this do something different?

Reply:
This adds two missing pieces: (a) Flattens out `app` and `protocol` fields (b) the result is in the format expected from older snapshots. See comments in the PR description. We could do the needed work (mentioned there) in the follow up PR which will remove the need for this function. Then sequence would be: (1) still handle extraction of `app` trees (2) call `convertSummaryTreeToITree`.

Reply:
sounds good. As long as this is temporary its fine :) Can you create a work item to remove this (if not already present).
Also, is it possible to move this to where it is used so we don't have to export it?

Reply:
https://github.com/microsoft/FluidFramework/issues/8979. Please add any labelling/projects so we have visibility here. I'll take a look if this can be done as a quick (separate from this PR) effort. It seems like it should. Otherwise, I'll add it to the backlog.

This call is used at two different places (storage services), and it needs to be at a driver level so that layer checks don't fail. Let me know if there is a better place then driver utils. Anyways, I'm hopeful we can just get rid of it in the short term.